### ISSUE TITLE
sensor 'Found 0 items in Messages'

### ISSUE THREAD

**Issue by irom77** [StackStorm-Exchange/stackstorm-msexchange#8]
Despite action `st2 run msexchange.search_items folder=Inbox` finding new item in **Inbox**, sensor `msexchange.ItemSensor` is not finding it. See below , I added time and folder to log:

```
2018-02-24 12:48:58,107 INFO [-] Found 0 items in Messages (Inbox) at 2018-02-24 12:48:58.031213-05:00
2018-02-24 12:49:03,245 INFO [-] Found 0 items in Messages (Inbox) at 2018-02-24 12:49:03.156893-05:00
2018-02-24 12:49:08,395 INFO [-] Found 0 items in Messages (Inbox) at 2018-02-24 12:49:08.297001-05:00

```

Not sure if it is important but I can see one error
```
$ sudo /opt/stackstorm/st2/bin/st2sensorcontainer --config-file=/etc/st2/st2.conf --sensor-ref=msexchange.ItemSensor
2018-02-24 15:08:22,306 DEBUG [-] Using config files: /etc/st2/st2.conf
...
2018-02-24 15:08:23,835 ERROR [-] Exception in thread Thread-17:
Traceback (most recent call last):
  File "/usr/lib/python2.7/threading.py", line 801, in __bootstrap_inner
    self.run()
  File "/usr/lib/python2.7/threading.py", line 754, in run
    self.__target(*self.__args, **self.__kwargs)
  File "/usr/lib/python2.7/multiprocessing/pool.py", line 325, in _handle_workers
    while thread._state == RUN or (pool._cache and thread._state != TERMINATE):
AttributeError: '_MainThread' object has no attribute '_state'

2018-02-24 15:08:23,836 ERROR [-]
2018-02-24 15:09:50,719 INFO [-] Running sensor in passive mode
2018-02-24 15:09:50,729 AUDIT [-] Access granted to "sensor_service" with the token set to expire at "2018-02-25T20:09:50.726292Z". (username='sensor_service',token_expiration='2018-02-25T20:09:50.726292Z')
2018-02-24 15:09:50,730 WARNING [-] "auth.api_url" configuration option is not configured

```

**Comment by LindsayHill** [comment]
Looking through the code, it will only create triggers for emails received after the sensor has started. Has that account received any new messages since the sensor was started?

**Comment by irom77** [comment]
Correct, that's my understanding too. I've sent and received many emails after the sensor was started

**Comment by LindsayHill** [comment]
Looks like there is something eventlet-related going on. See https://github.com/StackStorm/st2/issues/3818

**Comment by irom77** [comment]
Is that sensor code correct at all ? I am getting items found with i.e.  
```
start_date = timezone.localize(EWSDateTime.now()) - timedelta(minutes=1)
```

**Comment by LindsayHill** [comment]
There were some eventlet fixes that went into 2.7.2 - this may now be fixed https://github.com/StackStorm/st2/issues/3498#issuecomment-390951117

**Comment by ngesnetworks** [comment]
I'm seeing the same issue with the polling sensor generating the `INFO [-] Found 0 items`, yet I can run `st2 run msexchange.search_items folder=Inbox` and have it finding new items in the Inbox, the sensor `msexchange.ItemSensor` is not finding/triggering on any new items.

```2018-07-31 12:40:54,025 DEBUG [-] Using Python: 2.7.12 (/opt/stackstorm/st2/bin/python)
2018-07-31 12:40:54,026 DEBUG [-] Using config files: /etc/st2/st2.conf
2018-07-31 12:40:54,026 DEBUG [-] Using logging config: /etc/st2/logging.sensorcontainer.conf
2018-07-31 12:40:54,030 INFO [-] Connecting to database "st2" @ "127.0.0.1:27017" as user `"stackstorm".
2018-07-31 12:40:54,035 INFO [-] Successfully connected to database "st2" @ "127.0.0.1:27017" as user "stackstorm".
2018-07-31 12:40:54,276 INFO [-] Running in single sensor mode, using a single sensor partitioner...
2018-07-31 12:40:54,280 INFO [-] Setting up container to run 1 sensors.
2018-07-31 12:40:54,281 INFO [-]        Sensors list - [u'msexchange.ItemSensor'].
2018-07-31 12:40:54,281 INFO [-] (PID:7460) SensorContainer started.
2018-07-31 12:40:54,283 INFO [-] Running sensor msexchange.ItemSensor
2018-07-31 12:40:54,289 AUDIT [-] Access granted to "sensors_container" with the token set to expire at "2018-08-01T17:40:54.286474Z". (username='sensors_container',token_expiration='2018-08-01T17:40:54.286474Z')
2018-07-31 12:40:54,289 WARNING [-] "auth.api_url" configuration option is not configured
2018-07-31 12:40:54,344 INFO [-] Connected to amqp://guest:**@127.0.0.1:5672//
2018-07-31 12:40:54,370 INFO [-] Sensor msexchange.ItemSensor started
2018-07-31 12:40:55,766 INFO [-] Found config for sensor "ItemSensor"
2018-07-31 12:40:55,794 INFO [-] Watcher started
2018-07-31 12:40:55,795 INFO [-] Running sensor initialization code
2018-07-31 12:40:55,818 INFO [-] Connected to amqp://guest:**@127.0.0.1:5672//
2018-07-31 12:40:56,310 INFO [-] Running sensor in passive mode
2018-07-31 12:40:56,312 WARNING [-] "auth.api_url" configuration option is not configured
2018-07-31 12:40:56,319 AUDIT [-] Access granted to "sensor_service" with the token set to expire at "2018-08-01T17:40:56.316871Z". (username='sensor_service',token_expiration='2018-08-01T17:40:56.316871Z')
2018-07-31 12:40:56,449 INFO [-] Found 0 items
2018-07-31 12:41:01,622 INFO [-] Found 0 items
```
I also validated by using `st2 trigger-instance list` and found that there was no trigger generated.

The ST2 version is `st2 2.8.1, on Python 2.7.12`
